tests/e2e/support/commands.ts

```diff
@@ -98,7 +151,15 @@ const grapherCommands = [
 ];
 
 grapherCommands.forEach(command => {
-  Cypress.Commands.add(`${command}Grapher`, (x, y) => {
+  Cypress.Commands.add(`${command}Grapher`, (x: number, y: number) => {
     return grapherMouseCommand(command, x, y);
   });
 });
+
+/**
+ * Gets the vuex store from window.app
+ */
+Cypress.Commands.add('getStore', () => {
```

Comment:
Whoa why are you getting vuex stores in cypress? Cypress should be an end to end test, meaning you write tests that interact with the website the way a user does.

Instead of checking something from the store, check whatever the user should see change

Reply:
For some of these tests, I'm treating them more like integration tests. What this is helping me do is set an initial state for an app by directly making mutations to the vuex store. I don't intend to check the store values in tests. An example of this is that I want to test the stuntsheet controls. Rather than setting up a bunch of actions that mock out creating multiple stuntsheets and dots, I can just create a `Show` object and make a commit `setShow` at the beginning of each test.

I think that this would reduce test code size, and also make tests less prone to flakiness since they are shorter. Also see this example from cypress on how they interact with the vuex store: https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/blogs__vue-vuex-rest/cypress/integration/store-spec.js

Is there a different way to set up an initial state for tests?

Reply:
hm i disagree with that example of using Cypress and Vue 😛 but sure, do what you want.

My opinion is that exposing the Vue store gives too much power to integration tests. E2E tests shouldn't care about the internal implementation of how and where things are stored. My preferred solution is to do one of the following:

1. Make a cypress helper for each part you need, e.g. `cy.addStuntsheet`, `cy.addDot`, etc. These shouldn't be used for the actual test (i.e. testing stuntsheets or testing dots, respectively), but for other tests that assume this functionality works. These helpers would then use the normal Cypress API to click the things and what not. I like this because it's still an end to end test of how users will do actions in a given workflow without going into implementation details, but it is slow. However, I would argue that speed should not be a concern for Cypress tests. The idea is to have lots of really fast unit tests, and cypress should just test the full workflow, that everything works well with each other. This method is similar to what we do at my work.

2. You can also write a single helper `cy.initShow()` that sets up (via the normal Cypress API) some preset show (e.g. 2 stuntsheets, 50 dots, etc.). That way, you can start working off of a fairly well set up show and start an integration test from the middle. This is good because it still uses the Cypress API to act as the user would, but has the downside of possibly doing more than a given test needs it to do. I would recommend not being able to pass in an arbitrary `Show` because that would leak implementation details (namely, the structure of the `Show` javascript class).

3. You can also do either of the above using `cy.getStore()` instead of going through the Cypress API. It would be faster, but in my opinion, this leaks implementation details the most. Cypress should ideally not even know you're using Vue

Reply:
:+1: Yeah, these suggestions make sense. I think the helper commands would go a long way for us! I'll make an issue to create helper functions.

Also, I just realized that we'll have functionality to load a show from a file soon :laughing:! I think that will be a good way to set up initial state!

Let's do `cy.getStore()` as the short-term fix until we get that going. We may need your help with setting up cypress tests that involve importing a show file later!